packages/base/src/features/patchPopup.ts

```diff
@@ -68,16 +79,56 @@ const hasWebComponentPopupAbove = (popup: object) => {
 	return false;
 };
 
-const openNativePopover = (domRef: HTMLElement) => {
+const enableNativePopoverForOpenUI5 = (domRef: HTMLElement, popup: OpenUI5Popup) => {
+	const openUI5BlockLayer = document.getElementById("sap-ui-blocklayer-popup");
+
+	if (popup.getModal() && openUI5BlockLayer) {
+		openUI5BlockLayer.setAttribute("popover", "manual");
+		openUI5BlockLayer.hidePopover();
+		openUI5BlockLayer.showPopover();
+	}
+
 	domRef.setAttribute("popover", "manual");
 	domRef.showPopover();
 };
 
-const closeNativePopover = (domRef: HTMLElement) => {
+const disableNativePopoverForOpenUI5 = (domRef: HTMLElement) => {
 	if (domRef.hasAttribute("popover")) {
 		domRef.hidePopover();
 		domRef.removeAttribute("popover");
 	}
+
+	const lastPopup = AllOpenedPopupsRegistry.openedRegistry[AllOpenedPopupsRegistry.openedRegistry.length - 1];
```

Comment:
no need to check for all of this. As you have the last popup when you call disableNativePopoverForOpenUI5?
it is enough to check if popup the last one and is modal? 

if you close a popup which is different then lastPopup, you should not hide the block layer. E.g. if dialog behind the last one is closed